### PR TITLE
Page format button

### PR DESCRIPTION
**Pull request by kalbermattenm**
This implements #122

Currently we are still missing the interaction on the feature once it has been drawn. To implement these ones is quite a challenge as there is nothing out of the box in the Openlayers library.

@maltaesousa, should I continue with this PR or do you think it would be good to already review this?

### CODE REVIEW

**Comment by maltaesousa** [comment]
This PR is ok, it's better to divide it in small PRs. I'll review this afternoon

File: front/src/assets/configs/config.json.tmpl [hunk 1]
```diff
@@ -43,5 +43,32 @@
...
+    {
+      "name": "A1",
+      "width": 2383.9,
+      "height": 1683.8
+    },
+    {
+      "name": "A0",
+      "width": 3370.4,
+      "height": 2383.9
+    }
```
Comment (maltaesousa) [inline front/src/assets/configs/config.json.tmpl:72]:
```suggestion
  "pageformats": [
    {
      "name": "A4",
      "width": 210,
      "height": 297
    },
    {
      "name": "A3",
      "width": 297,
      "height": 420
    },
    {
      "name": "A2",
      "width": 420,
      "height": 594
    },
    {
      "name": "A1",
      "width": 594,
      "height": 841
    },
    {
      "name": "A0",
      "width": 841,
      "height": 1189
    }
```

**Review by maltaesousa — CHANGES_REQUESTED** [review]

File: front/src/app/_services/map.service.ts [hunk 2]
```diff
@@ -542,4 +547,34 @@ export class MapService {
...
+    }
+    this.geocoderSource.clear();
+
+    const center = this.map.getView().getCenter();
+    const units = this.map.getView().getProjection().getUnits();
+
+    const INCHES_PER_METER_UNIT = 39.37;
+    const PI = 3.14159265359 ;
+    const w = format.width / 72 / INCHES_PER_METER_UNIT * scale / 2;
+    const h = format.height / 72 / INCHES_PER_METER_UNIT * scale / 2;
```
[the remark this answers is not in the document]
Comment (kalbermattenm) [reply front/src/app/_services/map.service.ts:563]: If you are sure about the `2000` factor, I am happy to merge...

File: front/src/app/welcome/map/map.component.ts [hunk 3]
```diff
@@ -142,4 +155,27 @@ export class MapComponent implements OnInit {
     this.mapService.toggleTracking();
   }
 
+  togglePageformat(): void {
+    const dialogRef = this.dialog.open(PageformatComponent, {
+      width: '250px',
```
[the remark this answers is not in the document]
Comment (kalbermattenm) [reply front/src/app/welcome/map/map.component.ts:160]: Actually, this is an official option of the Angular Material ` MatDialogConfig` documentation.

**Comment by kalbermattenm** [comment]
I got some conflicts on this branch. Resolving them now.

**Pull request merged**